lib/workload/stateless/filemanager/database/migrations/0002_add_s3_object_table.sql

```diff
@@ -24,6 +24,9 @@ create table s3_object (
     bucket text not null,
     -- The key of the object.
     key text not null,
+    -- The version id of the object. A 'null' string is used to indicate no version id. This matches logic in AWS which
+    -- also returns 'null' strings.
+    version_id text not null default 'null',
```

Comment:
I know this looks weird, but I think that it is okay because AWS specifies that when the `version_id` is the string `'null'`, then versioning is not enabled: https://docs.aws.amazon.com/AmazonS3/latest/userguide/versioning-workflows.html. I.e. effectively, versioning is always enabled, and AWS uses `'null'` when it's disabled.

There is an alternative, and that's to have another constraint, .e.g. `created_sequencer_unique`, `deleted_sequencer_unique` and `versioning_id_unique`, and then adding adding additional insert queries that target the `versioning_id_unique` for de-duplication, although I feel that this adds a lot more repetitive code.

Reply:
First option LGTM, weird but it's documented... I'd probably add that docs url in the comment itself for reference.